docs/manual/shared_memory.rst

```diff
@@ -1,54 +1,130 @@
 
 .. code-block:: bash
 
   sudo apt install cmake libacl1-dev libncurses5-dev pkg-config maven
 
-We use colcon to build iceoryx and Cyclone DDS.
+Next you will need to get and build iceoryx (all of this is assumed to occur in your home directory):
 
 .. code-block:: bash
 
-  mkdir -p ~/cyclone_iceoryx_ws/src
-  cd ~/cyclone_iceoryx_ws/src
-  git clone https://github.com/eclipse-cyclonedds/cyclonedds.git -b iceoryx
   git clone https://github.com/eclipse-iceoryx/iceoryx.git -b master
-  cd ~/cyclone_iceoryx_ws
-  colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release
+  cd iceoryx
+  cmake -Bbuild -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=install -DBUILD_ALL=True -DBUILD_SHARED_LIBS=ON -Hiceoryx_meta
+  cmake --build build --config Debug --target install
 
-***
-Run
-***
+After that, get Cyclone DDS (currently its iceoryx branch) and build it with shared memory support:
+
+.. code-block:: bash
+
+  git clone https://github.com/eclipse-cyclonedds/cyclonedds.git -b iceoryx
+  cd cyclonedds
+  cmake -Bbuild -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=install -DBUILD_EXAMPLES=On -DCMAKE_PREFIX_PATH=~/iceoryx/install/
+  cmake --build build --config Debug --target install
+
+When the compiler has finished, you should have built both iceoryx and Cyclone DDS. Their respective files can be found in the install directories which were created in the directories git has made for you.
+
+Configuration
+--------------
+
+There are two levels of configuration for Cyclone DDS with shared memory support, the shared memory service (iceoryx) level, and the Cyclone DDS level.
+
+The performance of the shared memory service is strongly dependant on how well its configuration matches up with the use cases it will be asked to support, and large gains in performance can be made by configuring it correctly.
+
+The memory of iceoryx is layed out as numbers of fixed-size segments, which taken together are iceoryx's memory pool. When a subscriber requests a block of memory from iceoryx, the smallest block which will fit the requested size and is available is provided to the subscriber from the pool.
+
+To ensure the best performance with the smallest footprint, the user is advised to configure iceoryx in such a manner that the memory pool only consists of blocks which will be useful to the exchanges to be done. Additionally, due to header information being sent along with the sample, the block size required from the pool is 64 bytes larger than the data type being exchanged. Lastly, iceoryx requires that the blocks be aligned to 4 bytes.
 
-Setup the environment before trying to run Cyclone DDS.
+Below follows an example of an iceoryx configuration file which has a memory pool of 2^15 blocks which can store data types of 16384 bytes (+ 64 byte header = 16448 byte block):
+
+.. code-block:: toml
+
+  [general]
+  version = 1
+
+  [[segment]]
+
+  [[segment.mempool]]
+  size = 16448
+  count = 32768
```

Comment:
The user must be aware that by using this kind of configuration iceorx will only be able to allocate memory of this size, i.e. larger allocations will fail (I think it throws an error if the required size does not even exist). In turn, smaller allocations may waste a lot of memory.  So currently it is advised to have a variety of sizes, e.g. powers of two or, if the type sizes are well known exactly those sizes ( + header size).

Reply:
You are correct on this point, but my fear is that we may drag in too much detail whose place should be the iceoryx documentation.

Maybe putting in as little detail as possible and linking to the relevant section in the iceoryx documentation for the rest is a good compromise?

Reply:
Sure. The user just needs be aware it is important for optimal use and where to find the information. That is the case here I think.